javascripts/discourse-kanban/connectors/extra-nav-item/kanban-connector.js

```diff
@@ -1,7 +1,8 @@
-import { displayConnector } from "../../lib/kanban-utilities";
+import { displayConnector, getCurrentCategoryFromUrl } from "../../lib/kanban-utilities";
 
 export default {
   shouldRender(args, component) {
-    return displayConnector(component.get("category.slug"));
+    const slug_fixed = (args.category && args.category.slug) || getCurrentCategoryFromUrl();
```

Comment:
I'm not keen on determining the category by parsing the `window.location`.

`args.category` should work in all cases. If it doesn't, then we need to fix the bug in core. Can you post some instructions which reproduce the bug?

Reply:
> I'm not keen on determining the category by parsing the .`window.location`
> 
> `args.category` should work in all cases. If it doesn't, then we need to fix the bug in core. Can you post some instructions which reproduce the bug?

I haven't found a wrong example yet; but the original solution `displayConnector(component.get("category.slug"))` must be invalid, always returning undefined.

As an amateur, in fact, I don't know how often discourse changes these core parameters. But I guess it is not more stable than parsing URL(as an alternative), otherwise these methods will not be invalid?

Of course, if you need to remove this alternative, I really welcome you to modify it to only get `categorySlug` from the api provided by discourse!

Reply:
Should that code snippet above work?

Reply:
> Should that code snippet above work?

Absolutely not. (
You should call the `api.category` method instead. But I didn't verify the circumstances under which it is feasible.